### PR TITLE
[#855] docs: adjust intro, title and README badges

### LINKED ISSUE #855: Proposed update for README intro
I think the README intro paragraph should be re-written https://github.com/typesafegithub/github-workflows-kt#readme

It starts off by talking about YAML, which I don’t think is a strong start. I’d expect it to start with a strong statement about what workflows-kt is and does before talking about what problems it solves. For example:

> [workflows-kt](https://github.com/typesafegithub/github-workflows-kt) is a tool for creating GitHub Workflows in a _typesafe_ script, helping you to build robust workflows for your GitHub projects without mistakes. You won’t go back to YAML!

(just an example to show my point, I think it should be expanded a little more)

* It's a good idea to have a link back to the repo to help [when the repo gets rehosted](https://sourcegraph.com/github.com/typesafegithub/github-workflows-kt)
* The catchphrase `"You won't get back to YAML!"` is good, but a little awkward in English (but maybe that's the point, and it's hard to judge if it's intended to be a little off :)) - `"You won't go back to YAML!"` flows better.

### PR DESCRIPTION
**Pull request by krzema12**
Closes #855.

### CODE REVIEW

**Comment by krzema12** [comment]
@aSemy does it look good?

**Comment by aSemy** [comment]
It looks good 👍. I have some more suggestions:

* Using the `>` quote is good to add some styling to the tagline, but it de-emphasises the intro.
* The "Idea" and "Benefits" sections in the documentation README give a really good overview to new visitors. These should be copied to be in the README too.
* Add direct links to the 'getting started' and 'how to contribute' sections in the docs 
* Expand on the benefits: While the existing benefits are good, consider adding more specific examples or use cases to highlight how the library improves workflows. This helps users understand the practical advantages of using Kotlin instead of YAML.
* Organize the "Benefits" section: The "Benefits" section is well-structured, but it could be improved by using bullet points or a table format to make it easier to read and scan.
* JSON is mentioned, but is that relevant here? GitHub Workflows only deals with solving YAML drawbacks. I think it can be dropped.
* Provide API documentation: link to API documentation that provides detailed information about the library's classes, methods, and functions (generate this using Dokka)


Here's a first-draft re-write to demonstrate:

    
````markdown
# github-workflows-kt

[github-workflows-kt](https://github.com/typesafegithub/github-workflows-kt/) is a tool for creating GitHub Actions workflows in a _type-safe_ script, helping you to build robust workflows for your GitHub projects without mistakes, with pleasure, in Kotlin.

> _You won't go back to YAML!_

### 💡 Idea

We're often surrounded by YAML configuration. It's a powerful format that provides simple syntax for defining hierarchical data, but it is sometimes used (abused?) to configure complicated scenarios which leads to complicated files that are difficult to write and maintain. Who among us hasn't accidentally used the wrong indentation, forgotten a trailing comma, or been confused by ambiguous types? The power of a high-level language is more suitable.

We're developing github-workflows-kt to solve these problems, so you can create GitHub Workflows with confidence.

### ✨ Benefits

* **no indentation confusion** - Kotlin's syntax doesn't rely on it
* **immediate validation** - catch bugs early during development, not during runtime
* **strongly typed values** - no more confusion about what argument a parameter needs
* **superb IDE support** - author your workflows in any IDE that supports Kotlin, with auto-completion and documentation at your fingertips
* **remove duplication** - don't repeat yourself! Share common configuration using constant values, or define your own functions to encapsulate logic
* **fully featured language** - call an API to generate workflows dynamically, randomly generate data, or add custom validation
````

**Comment by krzema12** [comment]
Thanks for this feedback, makes sense! I'll process it thoroughly within the next several days.

**Comment by krzema12** [comment]
@aSemy please take another look, I used most of your draft + made only slight adjustments.

File: README.md [hunk 1]
```diff
@@ -1,18 +1,50 @@
 ![Logo white](images/Logo-White.svg#gh-dark-mode-only)
 ![Logo white](images/Logo-Black.svg#gh-light-mode-only)
 
-> _You won't go back to YAML!_
-
 <!-- maven coordinates -->
 [![Maven Central](https://maven-badges.herokuapp.com/maven-central/io.github.typesafegithub/github-workflows-kt/badge.svg)](https://maven-badges.herokuapp.com/maven-central/io.github.typesafegithub/github-workflows-kt)
 [![Slack channel](https://img.shields.io/badge/chat-slack-blue.svg?logo=slack)](https://kotlinlang.slack.com/messages/github-actions/)
 [![Awesome Kotlin Badge](https://kotlin.link/awesome-kotlin.svg)](https://github.com/KotlinBy/awesome-kotlin)
```
Comment (LeoColman) [inline README.md:8]:
```suggestion
[![GitHub Repo stars](https://img.shields.io/github/stars/typesafegithub/github-workflows-kt)](https://star-history.com/#typesafegithub/github-workflows-kt&Date)
[![License](https://img.shields.io/github/license/typesafegithub/github-workflows-kt)](https://github.com/typesafegithub/github-workflows-kt/blob/main/LICENSE)
[![github-workflows-kt @ kotlinlang.slack.com](https://img.shields.io/static/v1?label=kotlinlang&message=github-workflows-kt&color=blue&logo=slack)](https://kotlinlang.slack.com/archives/C02UUATR7RC)
```

**Review by LeoColman — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -1,18 +1,52 @@
 ![Logo white](images/Logo-White.svg#gh-dark-mode-only)
 ![Logo white](images/Logo-Black.svg#gh-light-mode-only)
 
-> _You won't go back to YAML!_
-
 <!-- maven coordinates -->
 [![Maven Central](https://maven-badges.herokuapp.com/maven-central/io.github.typesafegithub/github-workflows-kt/badge.svg)](https://maven-badges.herokuapp.com/maven-central/io.github.typesafegithub/github-workflows-kt)
 [![Slack channel](https://img.shields.io/badge/chat-slack-blue.svg?logo=slack)](https://kotlinlang.slack.com/messages/github-actions/)
```
Comment (krzema12) [inline README.md:6]:
```suggestion
```

File: README.md [hunk 3]
```diff
@@ -1,18 +1,51 @@
...
-> _You won't go back to YAML!_
-
 <!-- maven coordinates -->
 [![Maven Central](https://maven-badges.herokuapp.com/maven-central/io.github.typesafegithub/github-workflows-kt/badge.svg)](https://maven-badges.herokuapp.com/maven-central/io.github.typesafegithub/github-workflows-kt)
-[![Slack channel](https://img.shields.io/badge/chat-slack-blue.svg?logo=slack)](https://kotlinlang.slack.com/messages/github-actions/)
 [![Awesome Kotlin Badge](https://kotlin.link/awesome-kotlin.svg)](https://github.com/KotlinBy/awesome-kotlin)
+[![GitHub Repo stars](https://img.shields.io/github/stars/typesafegithub/github-workflows-kt)](https://star-history.com/#typesafegithub/github-workflows-kt&Date)
+[![License](https://img.shields.io/github/license/typesafegithub/github-workflows-kt)](https://github.com/typesafegithub/github-workflows-kt/blob/main/LICENSE)
+[![github-workflows-kt @ kotlinlang.slack.com](https://img.shields.io/static/v1?label=kotlinlang&message=github-workflows-kt&color=blue&logo=slack)](https://kotlinlang.slack.com/archives/C02UUATR7RC)
+[github-workflows-kt](https://github.com/typesafegithub/github-workflows-kt/) is a tool for creating
```
Comment (krzema12) [inline README.md:10]:
```suggestion
[![github-workflows-kt @ kotlinlang.slack.com](https://img.shields.io/static/v1?label=kotlinlang&message=github-workflows-kt&color=blue&logo=slack)](https://kotlinlang.slack.com/archives/C02UUATR7RC)

[github-workflows-kt](https://github.com/typesafegithub/github-workflows-kt/) is a tool for creating
```

**Pull request merged**